tenzies/src/component/Main.jsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useRef, useState } from "react";
+import { nanoid } from "nanoid";
+import NumPad from "./NumPad.jsx";
+import Confetti from "react-confetti";
+
+export default function Main() {
+  function getRandList() {
+    return new Array(10).fill(0).map(() => ({
+      value: Math.ceil(Math.random() * 6),
+      id: nanoid(),
+      isHeld: false,
+    }));
+  }
+
+  const [numPad, setNumpad] = useState(() => getRandList());
+  const buttonRef = useRef(null);
+  const numPadList = numPad.map((pad) => (
+    <NumPad
+      key={pad.id}
+      value={pad.value}
+      isHeld={pad.isHeld}
+      hold={() => hold(pad.id)} // Pass as a function reference
```

Comment:
you are passing the hold function here bounded to `pad.id`, refer to my comment on `NumPad.tsx` for more information